src/components/ConnectionsListProfiles.tsx

```diff
@@ -68,17 +102,23 @@ const ConnectionsListProfiles = ({
           <Button
             className="ConnectionsListProfiles__button"
             name={userRelationship(userProfile)}
+            onClick={() => changeGraphState(userProfile)}
+            disabled={isFollowingUpdating(userProfile)}
           >
             {userRelationship(userProfile)}
-            <div
-              className={
-                userRelationship(userProfile) === "Follow"
-                  ? "ConnectionsListProfiles__buttonFollowIcon"
-                  : "ConnectionsListProfiles__buttonUnfollowIcon"
-              }
-            >
-              &#10005;
-            </div>
+            {isFollowingUpdating(userProfile) ? (
+              <Spin></Spin>
+            ) : (
+              <div
+                className={
+                  userRelationship(userProfile) === "Follow"
```

Comment:
Isn't this just `isFollowing`?

Reply:
These nested function calls make me wonder if this should be a separate component and you can just map over that with all the calls being just params going into the functional component.